lib/node_modules/@stdlib/repl/lib/completer_engine.js

```diff
@@ -0,0 +1,747 @@
+	var lastMatchedIndex;
+	var completionIndex;
+	var highlighted;
+	var boldIndexes;
+	var completion;
+	var lineIndex;
+	var i;
+	var j;
+
+	highlightedCompletions = [];
+	for ( i = 0; i < this._completionsList.length; i++) {
+		completion = this._completionsList[ i ];
+		lastMatchedIndex = -1;
+		completionIndex = 0;
+		highlighted = '';
+		boldIndexes = []; // Buffer to store indexes of characters in completion string that needs to be highlighted
+		lineIndex = 0;
+
+		// If input is an exact prefix of completion, directly highlight the substring...
+		if ( startsWith( completion, this._completionPrefix ) ) {
+			highlighted = '\u001b[1m' + completion.slice( 0, this._completionPrefix.length ) + '\u001b[0m' + completion.slice( this._completionPrefix.length );
+		} else {
+			// Store indexes of each matching character in the completion string in the buffer...
+			while ( lineIndex < this._completionPrefix.length && completionIndex < completion.length ) { // eslint-disable-line max-len
+				if ( lowercase( completion[ completionIndex ] ) === lowercase( this._completionPrefix[ lineIndex ] ) ) { // eslint-disable-line max-len
+					boldIndexes.push( completionIndex );
+					lastMatchedIndex = completionIndex;
+					lineIndex += 1;
+				} else if ( completionIndex + 1 === completion.length ) {
+					lineIndex += 1;
+					completionIndex = lastMatchedIndex + 1;
+				}
+				completionIndex += 1;
+			}
+			// Highlight stored indexes in the completion string:
+			for ( j = 0; j < completion.length; j++ ) {
+				if ( contains( boldIndexes, j ) ) {
+					highlighted += '\u001b[1m' + completion[ j ] + '\u001b[0m';
+				} else {
+					highlighted += completion[ j ];
+				}
+			}
+		}
+		highlightedCompletions.push( highlighted );
+	}
+	return highlightedCompletions;
+});
+
+/**
+* Returns the number of columns in the completions output grid.
+*
+* @name _completionColumns
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @returns {number} number of columns
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_completionColumns', function completionColumns() {
+	return floor( this._repl.viewportWidth() / this._widthOfColumn ) || 1;
+});
+
+/**
+* Returns the number of rows in the completions output grid.
+*
+* @name _completionRows
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @param {number} columns - number of columns in the completions output grid
+* @returns {number} number of rows
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_completionRows', function completionRows( columns ) {
+	var maxRows = max( this._repl.viewportHeight() - ( this._multiline.inputHeight() + RESERVED_COMPLETER_ROWS ), 0 ); // eslint-disable-line max-len
+	var rows = ceil( this._completionsList.length / columns );
+
+	// Truncate number of completion rows to fit the viewport:
+	return min( rows, maxRows );
+});
+
+/**
+* Checks whether content having a specified number of lines is able to fit within the current viewport.
```

Comment:
This description seems off. The method doesn't take any arguments, so saying a "specified number of lines" does not seem appropriate.